### PR TITLE
[Rust] Add few commands and refactor the code

### PR DESCRIPTION
**Pull request by PunitLodha**
**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/ccextractor/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.
- [ ] **I have mentioned this change in the [changelog](https://github.com/CCExtractor/ccextractor/blob/master/docs/CHANGES.TXT).**

**My familiarity with the project is as follows (check one):**

- [ ] I have never used CCExtractor.
- [ ] I have used CCExtractor just a couple of times.
- [ ] I absolutely love CCExtractor, but have not contributed previously.
- [x] I am an active contributor to CCExtractor.

 Added handlers for CLW, HDW, TGW, DLW, and CR, and did some refactoring

### CODE REVIEW

File: src/lib_ccx/ccx_decoders_common.c [hunk 1]
```diff
@@ -16,9 +16,7 @@ made to reuse, not duplicate, as many functions as possible */
 #include "ccx_dtvcc.h"
 
 #ifdef ENABLE_RUST
-// Include rust decoder
-extern void dtvcc_process_cc_data(struct dtvcc_ctx *dtvcc,
-				  const unsigned char *data);
+extern int dtvcc_process_cc_data(struct lib_cc_decode *dec_ctx, unsigned char *cc_data, int cc_count);
```
Comment (cfsmp3) [inline src/lib_ccx/ccx_decoders_common.c:19]: I think it's safe to have this declaration all the time, ENABLE_RUST or not :-)

File: src/lib_ccx/ccx_decoders_common.c [hunk 2]
```diff
@@ -202,9 +204,7 @@ int do_cb(struct lib_cc_decode *ctx, unsigned char *cc_block, struct cc_subtitle
 				{
 					if (ctx->write_format != CCX_OF_RCWT)
 					{
-#ifdef ENABLE_RUST
-						dtvcc_process_cc_data(ctx->dtvcc, (const unsigned char *)temp);
-#else
+#ifndef ENABLE_RUST
 						dtvcc_process_data(ctx->dtvcc, (const unsigned char *)temp);
```
Comment (cfsmp3) [inline src/lib_ccx/ccx_decoders_common.c:208]: Why do we have two separate functions? dtvcc_process_cc_data is the rust version now?

File: src/rust/src/lib.rs [hunk 3]
```diff
@@ -28,3 +34,80 @@ pub extern "C" fn init_logger() {
...
         .init();
 }
+
+/// Process cc_data
+///
+/// # Safety
+/// dec_ctx should not be a null pointer
+/// data should point to cc_data of length cc_count
+#[no_mangle]
+pub unsafe extern "C" fn dtvcc_process_cc_data(
```
Comment (cfsmp3) [inline src/rust/src/lib.rs:44]: I hate globally unsafe functions :-)

File: src/lib_ccx/ccx_decoders_common.c (hunk 1 above)
Comment (PunitLodha) [reply src/lib_ccx/ccx_decoders_common.c:19]: That's a rust function, so it needs to be inside ENABLE_RUST

File: src/lib_ccx/ccx_decoders_common.c (hunk 2 above)
Comment (PunitLodha) [reply src/lib_ccx/ccx_decoders_common.c:208]: I realize that the naming for rust function might be confusing. There's only a single rust function called inside C code, named dtvcc_process_cc_data

**Comment by cfsmp3** [comment]
Consider having a prefix for all rust functions

**Pull request merged**